src/Microsoft.SqlTools.ServiceLayer/Connection/ListDatabaseRequestHandler.cs

```diff
@@ -58,16 +58,21 @@ abstract class ListDatabaseRequestHandler<T> : IListDatabaseRequestHandler
         private static readonly string[] SystemDatabases = new string[] { "master", "model", "msdb", "tempdb" };
 
         public abstract string QueryText { get; }
-
+        
         public ListDatabasesResponse HandleRequest(ISqlConnectionFactory connectionFactory, ConnectionInfo connectionInfo)
         {
             ConnectionDetails connectionDetails = connectionInfo.ConnectionDetails.Clone();
 
-            // Connect to master
-            connectionDetails.DatabaseName = "master";
             using (var connection = connectionFactory.CreateSqlConnection(ConnectionService.BuildConnectionString(connectionDetails), connectionDetails.AzureAccountToken))
             {
                 connection.Open();
+
+                // Connect to master if not a contained user 
+                if (!connectionFactory.CheckIfContainedAuthorization(connection))
+                {
+                    connectionDetails.DatabaseName = "master";
```

Comment:
why are you forcing a connection to master in Azure?

Reply:
I though that line was present for back compatibility. If it is not necessary, I will just get rid of this line.

Reply:
this was my question as well.  I'd say if no one knows why we need to set to master, perhaps we should just not do that and test on the lowest supported server version for backcompat (SQL Server 2014).